### PR TITLE
motion: Add motion support

### PR DESCRIPTION
**Pull request by Macdu**
Add a way to emulate the PS Vita gyro / accelerometer using a gamepad gyro / accelerometer.
Not all functions are implemented but the main one are. I only tested it on gravity rush, there may be a few mistakes when converting vectors from the SDL to Vita dimension space.

### CODE REVIEW

File: vita3k/modules/SceDriverUser/SceMotion.cpp [hunk 1]
```diff
@@ -68,6 +77,18 @@ EXPORT(SceBool, sceMotionGetMagnetometerState) {
...
+    std::lock_guard<std::mutex> guard(emuenv.motion.mutex);
+    sensorState->accelerometer = get_acceleration(emuenv.motion);
+    sensorState->gyro = get_gyroscope(emuenv.motion);
+
+    sensorState->timestamp = emuenv.motion.last_accel_timestamp;
+    sensorState->counter = emuenv.motion.last_counter;
+    sensorState->hostTimestamp = sensorState->timestamp;
+    sensorState->dataInfo = 0;
+
     return UNIMPLEMENTED();
```
Comment (EXtremeExploit) [inline vita3k/modules/SceDriverUser/SceMotion.cpp:92]: forgor to change this to 0 or its just a stub? :thinking:

**Comment by ArbestRiAgain** [comment]
Does this work on controllers with motion support?

**Comment by ArbestRiAgain** [comment]
Works really well with Gravity Rush and LBPV!

**Comment by Zangetsu38** [comment]
have probleme on flower/flow and wipeout

File: vita3k/motion/src/motion.cpp [hunk 2]
```diff
@@ -19,4 +19,111 @@
...
+    Util::Vec3f gyroscope = state.motion_data.GetGyroscope() * static_cast<float>(2.f * M_PI);
+    return {
+        gyroscope.x,
+        gyroscope.y,
+        gyroscope.z,
+    };
+}
+
+SceFVector3 get_gyro_bias(const MotionState &state) {
+    Util::Vec3f bias = state.motion_data.GetGyroscopeBias();
```
Comment (german77) [inline vita3k/motion/src/motion.cpp:45]: You converted gyro input into degrees. Shouldn't this be converted as well since the bias value is in radians?

**Comment by v-fox** [comment]
This seems to be intended to be used while holding DualShock parallel to your body which is quite exhausting. When used perpendicularly, with gamepad resting on lap, [counter-]clockwise movement are reversed for x and y axes. Don't remember that happening with motion controls on cemu & yuzu. Maybe there is some kind of auto-switch function or they are designed for that type of hold.

**Review by Zangetsu38 — APPROVED** [review]
LGTM

**Comment by AndeanPuppy** [comment]
Hey, apologies for the necropost -- will there be an option to use the device itself as the gyroscope sensor in the Android builds? I'm not sure if it's there because tilting the smartphone by itself with no gamepad other than the touchscreen doesn't do anything in LBPV.

**Comment by Mahim2000** [comment]
Will the PC Version have it as well via. Keyboard Buttons? I was thinking of using the Numpad for Gyro Emulation.

**Pull request merged**